engine/storage/src/org/apache/cloudstack/storage/image/BaseImageStoreDriverImpl.java

```diff
@@ -79,6 +80,25 @@
     VMTemplateZoneDao _vmTemplateZoneDao;
     @Inject
     AlertManager _alertMgr;
+    @Inject
+    ImageStoreDetailsDao _imageStoreDetailsDao;
+
+    private static final String NFS_VERSION_DETAILS_KEY = "nfs.version";
```

Comment:
@koushik-das sorry for late reply. A reason for defining nfs details in `BaseImageStoreDriverImpl` was that as `CloudStackImageStoreDriverImpl` and `SimulatorImageStoreDriverImpl` both extend it and they both return a NfsTO instance on `getStoreTO` we would need that key at least for the 2 mentioned classes. What would you suggest?